### PR TITLE
Feat all the Api for the leaveRequest

### PR DESCRIPTION
**Pull request by Sameerthe-Big-O**

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -12,12 +12,15 @@ import permissionRoute from "./permission/index.js";
 
 import employeeRoute from "./employee/index.js";
 
+import leaveRequest from "./leaveRequest/index.js";
+
```
Comment (AbubakarZorrain) [inline routes/index.js:16]: Remove empty line.

File: routes/index.js [hunk 2]
```diff
@@ -12,12 +12,15 @@ import permissionRoute from "./permission/index.js";
...
 const protectedRouter = express.Router();
 const unProtectedRouter = express.Router();
 
 // Protected Routes
```
Comment (AbubakarZorrain) [inline routes/index.js:20]: Why this comment?

File: routes/index.js [hunk 3]
```diff
@@ -12,12 +12,15 @@ import permissionRoute from "./permission/index.js";
...
 
 // Un-Protected Routes
+unProtectedRouter.use("/leaveRequest", leaveRequest);
```
Comment (AbubakarZorrain) [inline routes/index.js:23]: Why leave route is unprotected?

File: routes/leaveRequest/controllers.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+
+
+import leaveRequest from "../../services/leaveRequest.js";
+import httpResponse from "../../utils/httpResponse.js";
+
+const controller = {
+  add: async (req, res) => {
+    try{
+      console.log(req.path.file);
```
Comment (AbubakarZorrain) [inline routes/leaveRequest/controllers.js:9]: Remove console.

File: routes/leaveRequest/controllers.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+        return httpResponse.SUCCESS(res,addResponse.data)
+      }
+    }
+    catch(addResponse) {
+       return httpResponse.INTERNAL_SERVER_ERROR(res,addResponse.data)
+  }
+},
+  
+
+  get: async (req, res) => {
```
Comment (AbubakarZorrain) [inline routes/leaveRequest/controllers.js:24]: Rename it to getOneById.

File: routes/leaveRequest/controllers.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+
+  getAll: async (req, res) => {
+    try {
+      const data = await leaveRequest.getAll();
+      return httpResponse.SUCCESS(res, data.data);
+    } catch (error) {
+      return httpResponse.INTERNAL_SERVER_ERROR(res, error.data);
+    }
+  },
+  update:async (req,res) => {
```
Comment (AbubakarZorrain) [inline routes/leaveRequest/controllers.js:47]: Rename it to updateOneById

File: routes/leaveRequest/controllers.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+      else{
+        return httpResponse.SUCCESS(res,data.data);
+      }
+    }
+    catch(error){
+      return httpResponse.INTERNAL_SERVER_ERROR(res, error.data);
+    }
+  },
+
+  delete:async (req,res) => {
```
Comment (AbubakarZorrain) [inline routes/leaveRequest/controllers.js:64]: Rename it to deleteOneById

File: services/leaveRequest.js [hunk 8]
```diff
@@ -0,0 +1,74 @@
+import leaveRequest from "../models/leaveRequest.js";
+
+const hierarchy = {
```
Comment (AbubakarZorrain) [inline services/leaveRequest.js:3]: why hierarchy is there in leaveRequest.js

File: services/leaveRequest.js [hunk 9]
```diff
@@ -0,0 +1,74 @@
...
+  getAll: async () => {
+    try {
+      const data = await leaveRequest.find();
+      return { data };
+    } catch (error) {
+      return { error };
+    }
+  },
+
+  get: async (id) => {
```
Comment (AbubakarZorrain) [inline services/leaveRequest.js:13]: Rename it accordingly.

File: services/leaveRequest.js [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      };
+
+      const data = await leaveRequest.create(file);
+      return { data };
+    } catch (error) {
+      return { data: error.message };
+    }
+  },
+
+  update: async (id, body, path) => {
```
Comment (AbubakarZorrain) [inline services/leaveRequest.js:42]: Rename it accordingly.

File: services/leaveRequest.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+      };
+
+      const data = await leaveRequest.findByIdAndUpdate(id, file);
+      return { data };
+    } catch (error) {
+      return { data: error.message };
+    }
+  },
+
+  delete: async (id) => {
```
Comment (AbubakarZorrain) [inline services/leaveRequest.js:62]: Rename it accordingly.

File: routes/index.js (hunk 3 above)
Comment (Sameerthe-Big-O) [reply routes/index.js:23]: Aoa sir, so we're just following the boilerplate however if i've searched it seems to rouets those that need authorization are actually protected but sir in that all the most of the routes will get protected

File: routes/index.js (hunk 2 above)
Comment (Sameerthe-Big-O) [reply routes/index.js:20]: the comment is sir for the future incoming protected route

File: routes/index.js (hunk 3 above)
Comment (AbubakarZorrain) [reply routes/index.js:22]: All the routes must be protected except login and signUp. @Azmat7860 @Sameerthe-Big-O @abubakar105

Comment (Azmat7860) [reply routes/index.js:22]: Ok Sir

**Comment by AbubakarZorrain** [comment]
Resolve conflicts...

**Pull request merged**